cranelift-codegen/meta/src/isa/x86/recipes.rs

```diff
@@ -1002,6 +1002,22 @@ pub(crate) fn define<'shared>(
             ),
     );
 
+    // XX+rd id unary with 32-bit immediate.
+    {
+        let format = formats.get(f_unary_imm);
+        recipes.add_template_recipe(
+            EncodingRecipeBuilder::new("u_id_z", f_unary_imm, 1)
+                .operands_out(vec![gpr])
+                .inst_predicate(InstructionPredicate::new_is_zero_int(format, "imm"))
```

Comment:
Ok, I know this is confusing (this confuses me at least), but one of the instruction predicates here or in the encodings instruction predicate is redundant, since they have the same effect. It's better to only keep one of them, until we figure out how to handle this better at the meta crate level (probably just keeping encodings predicates could be sufficient). I'd suggest to remove this one, because the encodings code generator is able to group encodings with the same predicates (removing checks in the legalizer interpreter).

Reply:
Ah, the redundant predicate bit was a holdover from updating the initial attempt from the old python meta code. I was also confused about that :) I've removed this predicate in favor of the ones in encodings.